### PR TITLE
 [tests] add ipv6 source selection test

### PR DESCRIPTION
**Pull request by librasungirl**
travis fail is expected before #4377 gets merged

verified pass with #4377 locally.

### CODE REVIEW

**Comment by jwhui** [comment]
I just merged #4377 and rebased this PR on top.

**Comment by librasungirl** [comment]
@jwhui Thanks :)

File: tests/scripts/thread-cert/test_ipv6_source_selection.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        # Set lowpan context of sniffer
+        self.simulator.set_lowpan_context(1, '2001::/64')
+
+        # flushes message queue before next ping
+        router_msgs = self.simulator.get_messages_sent_by(ROUTER)
+
+        # source check - GUA source for GUA destination
+        leader_gua = self.nodes[LEADER].get_addr("2001::/64")
+        router_gua = self.nodes[ROUTER].get_addr("2001::/64")
+        print(leader_gua)
```
Comment (LuDuda) [inline tests/scripts/thread-cert/test_ipv6_source_selection.py]: nit: remove it?

File: tests/scripts/thread-cert/test_ipv6_source_selection.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        print(router_gua)
+        self.assertTrue(leader_gua is not None)
+        self.assertTrue(router_gua is not None)
+        self.assertTrue(self.nodes[ROUTER].ping(leader_gua))
+        router_msgs = self.simulator.get_messages_sent_by(ROUTER)
+        msg = router_msgs.get_icmp_message(ipv6.ICMP_ECHO_REQUEST)
+        msg.assertSentFromSourceAddress(router_gua)
+
+        # source check - GUA source for external address (default route)
+        self.nodes[ROUTER].ping(external_addr)
```
Comment (LuDuda) [inline tests/scripts/thread-cert/test_ipv6_source_selection.py]: not: add assert as above

File: tests/scripts/thread-cert/test_ipv6_source_selection.py (hunk 1 above)
Comment (librasungirl) [reply tests/scripts/thread-cert/test_ipv6_source_selection.py]: removed. thanks.

File: tests/scripts/thread-cert/test_ipv6_source_selection.py (hunk 2 above)
Comment (librasungirl) [reply tests/scripts/thread-cert/test_ipv6_source_selection.py]: Different from the other pings, the ping here doesn't has the reply.

**Pull request merged**